### PR TITLE
CLOUD-3395 - EAP CD 18 release - correct docs / templates

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.jboss.org/browse/CLOUD-3395
Signed-off-by: Ken Wills <kwills@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by luck3y** [comment]
@jmesnil We had some random breakage with the docs generation, so I've had to use python2 for running gen_template_docs.py for now, I'll open a separate issue for this on py3. There was also a 17->18 missed in the SSO template.

**Comment by luck3y** [comment]
Once this is merged we should also merge the eap-cd-dev branch into eap-cd, as thats the legacy way of releasing images so any old references will be updated. (For most things we'll just use the CD18 tag.)

**Review by jfdenise — APPROVED** [review]

File: gen_template_docs.py [hunk 1]
```diff
@@ -235,7 +235,7 @@ def createObjectTable(data, tableKind):
             s2i = obj["spec"]["strategy"]["sourceStrategy"]["from"]["name"]
             link = " link:" + LINKS[s2i]
          elif obj["spec"]["strategy"]["type"] == 'Docker':
-            s2i = obj["spec"]["strategy"]["dockerStrategy"]["dockerfilePath"]
+            s2i = obj["spec"]["strategy"]["dockerStrategy"]["from"]["name"]
```
Comment (jmesnil) [inline gen_template_docs.py:238]: is that change related to using py2 instead of py3 as you commented?

**Comment by luck3y** [comment]
No, that's due to us changing the strategy in the templates from source to
docker the support the runtime image, I suspect the docs script had never
been tested with that previously, or the expected values were different.

**Review by jmesnil — APPROVED** [review]
LGTM

**Pull request merged**